staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/config/apis/webhookadmission/types.go

```diff
@@ -25,5 +25,36 @@ type WebhookAdmission struct {
 	metav1.TypeMeta
 
 	// KubeConfigFile is the path to the kubeconfig file.
+	// +optional
 	KubeConfigFile string
+
+	// ExclusionRules are a list of rules that the webhooks while ignore
+	// +optional
+	ExclusionRules []ExclusionRule //NOTE: using ExclusionRule rather than Rule with name/namespace because don't want to have scope field
```

Comment:
> I appreciate the usefulness of this from a cluster fleet management perspective.

The ability to prevent interference with core operations is a pretty important use case. In particular, it relieves the user from the need to keep track of when the set of critical resources changes.

> But it does have some surprising behavior.

Agreed, surprise should be minimized.

> This would force the webhook to be rewritten to not match the resources being excluded, but once modified, the webhook would behave exactly as expected.

We would still need to augment admission webhooks to have exclude rules (similar to your mutation solution), in order to support "match all objects except"-type validations, which are useful for metadata-focued admission webhooks.

> The mutating policy could modify any webhooks and add the appropriate exclusions

I like this idea in principle, but it depends on two assumptions:

* Having mutation rules that are separately authored by multiple parties doesn't pose a risk to idempotence.

* The exclusion criteria schema is formed in such a way that mutation rules can reliably add extra exclusions, regardless of the inbound object (adding flags to a command line argument is a counter example where this isn't necessarily true)

Also, to prevent tampering, platform providers would need to somehow make these mutator configs read only.

I think the biggest feature of the mutation and validation based approaches is that it forces the ignores to show up on the webhook config object, giving the user visibility. Could we do the same thing by adding a `status` entry to webhook configs that shows the exclusions?